backend/tests/integration/controllers/posts-controller.test.js

```diff
@@ -0,0 +1,155 @@
+import mongoose from 'mongoose';
+import request from 'supertest';
+import Post from '../../../models/post.js';
+import server from '../../../server.js';
+import { validCategories } from '../../../utils/constants.js';
+import { createPostObject } from '../../utils/helper-objects.js';
+
+afterAll(async () => {
+  await mongoose.disconnect();
+});
+
+let postId;
+const invalidPostId = '609c16c69405b14574c99999';
+describe('Integration Tests: Post creation', () => {
+  it('Post creation: Success - All fields are valid', async () => {
+    const response = await request(server).post('/api/posts').send(createPostObject());
```

Comment:
Arrange the expected or mocked data/objects/others together at start of better mind-model.

```sh
Arrange Act Assert (AAA)
```

Ref[AAA]: https://testing-library.com/docs/react-testing-library/example-intro#quickstart